### PR TITLE
feat: Implement complete CRUD API for Task Manager with full test coverage

### PR DESCRIPTION
**Pull request by codeByVaibhav**
## 🚀 Overview
Implemented a complete REST API for task management from scratch, transforming a basic Express.js server setup into a fully functional CRUD application that passes all test requirements.

## ✨ Features Added
- **Complete CRUD Operations**: Created all five essential endpoints for task management
- **Data Validation**: Implemented robust input validation with proper error handling
- **JSON Data Management**: Added file-based data persistence using `task.json`
- **HTTP Status Codes**: Proper status code implementation (200, 201, 400, 404)
- **Test Coverage**: All 19 test assertions passing

## 🔧 Technical Implementation
- **Framework**: Express.js with middleware setup
- **Data Storage**: In-memory data with JSON file initialization
- **Validation**: Field validation for required properties and data types
- **Error Handling**: Comprehensive 404 and 400 error responses
- **ID Management**: Auto-incrementing task IDs for new tasks

## 🧪 Testing
- ✅ All 19 test assertions pass
- ✅ Complete test coverage for all endpoints
- ✅ Edge case validation (invalid data, missing tasks)

## 📋 Files Changed/Added
- `app.js` - Complete API implementation
- `README.md` - Updated documentation

## 🎯 Test Results

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,10 +1,79 @@
 const express = require('express');
+var fs = require('fs');
```
[the remark this answers is not in the document]
Comment (codeByVaibhav) [reply app.js:2]: Ok will update
